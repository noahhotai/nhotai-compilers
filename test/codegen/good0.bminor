denis: string = "9";

main: function integer () = {
    i: integer = 10;
    z: integer = i;
    print z;
}

