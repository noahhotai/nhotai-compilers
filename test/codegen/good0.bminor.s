.file "test/codegen/good0.bminor.s"
.data
denis: 
.string "9"
.text
.global main
main:
PUSHQ %rbp
MOVQ %rsp, %rbp
PUSHQ %rdi
PUSHQ %rsi
PUSHQ %rdx
PUSHQ %rcx
PUSHQ %r8
PUSHQ %r9
MOVQ $10, %rbx
PUSHQ %rbx
MOVQ -56(%rbp), %rbx
PUSHQ %rbx
MOVQ -64(%rbp), %rbx
MOVQ %rbx, %rdi
CALL print_integer
ADDQ $16, %rsp
POPQ %r9
POPQ %r8
POPQ %rcx
POPQ %rdx
POPQ %rsi
POPQ %rdi
POPQ %rbp
RET
