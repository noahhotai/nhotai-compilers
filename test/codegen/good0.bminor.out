Program parsed successfully.
i resolves to local 0
z resolves to local 1
Program resolved successfully.
Program type checked successfully.
